### ISSUE TITLE
fmt v8 makes recent Xcode clang crash on C++20 mode

### ISSUE THREAD

**Issue by jcelerier** [fmtlib/fmt#2521]
Repro: 
```C++
#include <fmt/format.h>

template<typename... Args>
void foo(Args&&... args)
{
  fmt::format("{} [{}({})]", "kiki", "foo", 1234);
}

void bar()
{
   foo(1, 2, 3);
}
```

Building: 
```bash
$ clang++ -std=c++20 -o foo.o -c foo.cpp  -I fmt/include
clang: error: unable to execute command: Segmentation fault: 11
clang: error: clang frontend command failed due to signal (use -v to see invocation)
Apple clang version 13.0.0 (clang-1300.0.29.3)
Target: x86_64-apple-darwin20.5.0
Thread model: posix
InstalledDir: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin
clang: note: diagnostic msg:
********************

PLEASE ATTACH THE FOLLOWING FILES TO THE BUG REPORT:
Preprocessed source(s) and associated run script(s) are located at:
clang: note: diagnostic msg: /var/folders/_q/jdkbv4k94dg4y82p7pn3fz440000gn/T/foo-627636.cpp
clang: note: diagnostic msg: /var/folders/_q/jdkbv4k94dg4y82p7pn3fz440000gn/T/foo-627636.sh
clang: note: diagnostic msg: Crash backtrace is located in
clang: note: diagnostic msg: /Users/jcelerier/Library/Logs/DiagnosticReports/clang_<YYYY-MM-DD-HHMMSS>_<hostname>.crash
clang: note: diagnostic msg: (choose the .crash file that corresponds to your crash)
clang: note: diagnostic msg:

********************
```

not sure there's a lot fmt can do here but it's been happening since Xcode 12.3 and it still happens with Xcode 13 so if there's any idea of how to make it work somehow it would be great :(

causes https://github.com/gabime/spdlog/issues/2126

**Comment by jcelerier** [comment]
works on master tho

**Comment by brad-anderson** [comment]
I haven't confirmed that this is fixed in fmt 8.1.0 (which isn't yet on vcpkg) but I just thought I'd add a note for anyone else having this issue that it was still present in 8.0.1 running Apple clang version 13.0.0 (clang-1300.0.29.30).

**Comment by brad-anderson** [comment]
8.1.0 does indeed fix this issue.
